src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelInternal.java

```diff
@@ -258,17 +264,23 @@ void invalidate() {
   /** @return a boolean to indicate whether the channel is closed or not */
   @Override
   public boolean isClosed() {
-    return this.isClosed;
+    return this.terminationFuture.isDone();
```

Comment:
Looks like KC is using it, I'm fine with both approach depends on whether KC has a workaround

Reply:
I see, then I can leave it given that people are using it and also now it is safer :)